### PR TITLE
CPLAT-7476: Run tests with dart2js

### PR DESCRIPTION
**Pull request by olesiathoms-wk**
## Problem
We are getting ready to ship Dart2 code to production.Since we compile that code with dart2js, it would be great to run unit tests with dart2js as well. This will provide more confidence in the shipped code.
## Solution
- Update dependencies
- Update `.gitignore`
- Use pub run test
- Use pub2 solver for .travis.yml
## Testin instructions
- [ ] CI should be sufficient

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .travis.yml [hunk 1]
```diff
@@ -2,9 +2,10 @@ language: dart
...
   - 1.24.3
   - stable
+before_install:
+  - ./pub2get.sh
 script:
-  - pub get
   - pub run dart_dev format --check
   - pub run dart_dev analyze
-  - pub run dart_dev test
+  - pub run test
```
Comment (todbachman-wf) [inline .travis.yml]: Why this change? Should we specify the unit test file to run?

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,9 @@
+FROM google/dart:2.5 as dart2
+
 FROM google/dart:1.24.3
 WORKDIR /build/
 ADD pubspec.yaml /build/
-RUN pub get
+COPY --from=dart2 /usr/lib/dart /usr/lib/dart2
+RUN _PUB_TEST_SDK_VERSION=1.24.3 /usr/lib/dart2/bin/pub get --no-precompile
```
Comment (todbachman-wf) [inline Dockerfile]: Usually we do a second `pub get` (using the Dart 1 pub solver) immediately after this. Should we do that here?

File: .travis.yml (hunk 1 above)
Comment (olesiathoms-wk) [reply .travis.yml]: `pub run test` seemed cleaner compare to  `pub run dart_dev test --release -p chrome`. Both of them work, I can change it `pub run dart_dev test --release -p chrome` if that we prefer.

Comment (todbachman-wf) [reply .travis.yml]: The reason I asked is I noticed there is a generated runner in this repo. Since we're not specifying a test file to run it seems like it's going to run all the test files it finds which will result in all of the tests running twice.

I assumed ddev test was configured to run just the generated runner, but it appears there is no configuration for it in the dev.dart file. So when we were running ddev test, I'm not sure what it was doing.

Either way, it seems like it would be nice to at least specify the generated runner as the suite we want to run, whether thats on the command line when running pub run test or it is in the dev.dart test config. 🤷‍♂

File: Dockerfile (hunk 2 above)
Comment (olesiathoms-wk) [reply Dockerfile]: I thought we need to do second `pub get` only if we run into trouble with `pub2get` when it's unable to resolve `async` dependency. In this repo it resolves correctly, so I would say it's necessary here 🤷‍♀

File: .travis.yml (hunk 1 above)
Comment (todbachman-wf) [reply .travis.yml]: You know what? The generated runner doesn't end in `test.dart` so it is never run. I think we could just delete that file and call it good.

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,9 @@
+FROM google/dart:2.5 as dart2
```
Comment (jihongkim-wk) [inline Dockerfile]: Quick question, does this mean to run the tests in `Dart 2.5`?

Comment (olesiathoms-wk) [reply Dockerfile]: I had to add this line in Dockerfile, so would be able to use pub2 solver, otherwise pub get will timeout on dart1. 
Tests will run in dart2.5 tho, because we use `stable`: https://github.com/Workiva/r_tree/blob/master/.travis.yml#L4.

**Comment by travissanderson-wf** [comment]
+10 @Workiva/release-management-p
